sql/rowexec/rel.go

```diff
@@ -644,50 +644,66 @@ func (b *BaseBuilder) buildIndexedTableAccess(ctx *sql.Context, n *plan.IndexedT
 	return sql.NewSpanIter(span, sql.NewTableRowIter(ctx, n.Table, partIter)), nil
 }
 
-func (b *BaseBuilder) buildUnion(ctx *sql.Context, u *plan.Union, row sql.Row) (sql.RowIter, error) {
-	span, ctx := ctx.Span("plan.Union")
-	var iter sql.RowIter
+func (b *BaseBuilder) buildUnion(ctx *sql.Context, s *plan.SetOp, row sql.Row) (sql.RowIter, error) {
+	span, ctx := ctx.Span("plan.SetOp")
+	var iter, iter2 sql.RowIter
 	var err error
-	iter, err = b.buildNodeExec(ctx, u.Left(), row)
-
+	iter, err = b.buildNodeExec(ctx, s.Left(), row)
+	if err != nil {
+		span.End()
+		return nil, err
+	}
+	iter2, err = b.buildNodeExec(ctx, s.Right(), row)
 	if err != nil {
 		span.End()
 		return nil, err
 	}
-	iter = &unionIter{
-		cur: iter,
-		nextIter: func(ctx *sql.Context) (sql.RowIter, error) {
-			return b.buildNodeExec(ctx, u.Right(), row)
```

Comment:
So I think the reason why `nextIter` was a provider function was so that if the engine doesn't exhaust the first rowIter in a union, it doesn't need to initialize the second. We probably want to preserve that behavior.

Reply:
make sense to me